lib/src/main/kotlin/starknet/data/types/transactions.kt

```diff
@@ -24,12 +26,38 @@ enum class StarknetChainId(val value: Felt) {
     TESTNET(Felt.fromHex("0x534e5f474f45524c49")), // encodeShortString('SN_GOERLI'),
 }
 
-data class Invocation(
-    val contractAddress: Felt, val entrypoint: Felt, val calldata: Calldata?, val signature: Signature?
-)
+enum class BlockTag(val tag: String) {
+    LATEST("latest"),
+    PENDING("pending")
+}
+
+@Serializable
+sealed class BlockHashOrTag() {
+    data class Hash(
+        val blockHash: Felt
+    ) : BlockHashOrTag() {
+        override fun string(): String {
+            return blockHash.hexString()
+        }
+    }
+
+    data class Tag(
```

Comment:
Can't we place enum in here directly?

Reply:
Actually not, enums cannot inherit from classes.
What I did here was an attempt to achieve algebraic data types in kotlin, as described here: https://medium.com/sharenowtech/kotlin-adt-74472319962a